frame/pallet-xcm/src/tests.rs

```diff
@@ -591,6 +591,59 @@ fn unlimited_reserve_transfer_assets_works() {
     });
 }
 
+/// Test `reserve_withdraw_assets`
+///
+/// Asserts that the sender's balance is decreased and the beneficiary's balance
+/// is increased. Verifies the correct message is sent and event is emitted.
+#[test]
+fn reserve_withdraw_assets_works() {
+    let balances = vec![
+        (ALICE, INITIAL_BALANCE),
+        (ParaId::from(PARA_ID).into_account(), INITIAL_BALANCE),
+    ];
+    new_test_ext_with_balances(balances).execute_with(|| {
+        let weight = BaseXcmWeight::get();
+        let dest: MultiLocation = Junction::AccountId32 {
+            network: NetworkId::Any,
+            id: ALICE.into(),
+        }
+        .into();
+        assert_eq!(Balances::total_balance(&ALICE), INITIAL_BALANCE);
```

Comment:
Imo, it'd be good to also check the sovereign account initial and final balance.